app/controllers/clearance/passwords_controller.rb

```diff
@@ -24,10 +27,12 @@ def new
   end
 
   def update
-    @user = find_user_for_update
+    @password_reset = find_password_reset_by_user_id_and_token
+    user = @password_reset.user
 
-    if @user.update_password password_reset_params
-      sign_in @user
+    if user.update_password password_reset_params
+      Clearance::PasswordResetDeactivator.new(user).run
```

Comment:
Great idea! I've actually never used ActiveRecord Transactions before, and this totally makes sense. I'll give it a go.